dbms/src/Storages/KVStore/Decode/PartitionStreams.cpp

```diff
@@ -224,7 +305,7 @@ static DM::WriteResult writeRegionDataToStorage(
                 keyspace_id,
                 table_id);
         }
-        return write_result;
+        return std::move(rw_ctx.write_result);
```

Comment:
do not add `std::move` in return statement.

Reply:
Why?

Reply:
The compiler can optimize this. And https://dassencio.org/103

Reply:
Yes, move semantics can break copy elision. However, a copy elision can happen if there is a copy-ctor.
The class here deleted the copy-ctor to explicitly disable copy. So there is no copy-elision here.

Check https://en.cppreference.com/w/cpp/language/copy_elision for the following statement

> This is an optimization: even when it takes place and the copy/move(since C++11) constructor is not called, it still must be present and accessible (as if no optimization happened at all), otherwise the program is ill-formed

Reply:
There is no need to add `std::move`(complier can help optimize), and it may break some optimization in some cases, so why add it?

Reply:
Because the class is defined to be like this, I mean its copy-ctor is disabled. IMO, we don't need to change the class, since disabling the copy-ctor is a very common practice. In fact we are planning do this to `ReadBuffer` in a previous PR.

It is **necessary** to add move here. It is not a optimization issue. You can remove the `std::move` and try compile again, the compiler would complain error. This is because returning a lvalue here requires copy logic, which is disabled.

Meanwhile, compilers can choose to optimize the code by move-elision.